src/form.tsx

```diff
@@ -209,24 +216,36 @@ export const Form = forwardRef(function Form(
 
           try {
             const response = await fetchData(state.data);
+            const isJson = response.headers
+              .get('content-type')
+              .startsWith('application/json');
+
+            const data = isJson ? await response.json() : undefined;
 
             if (response.ok) {
               setState({
                 ...state,
                 state: 'success',
-                data: await response.json(),
+                data,
                 redirect: response.redirected ? response.url : undefined,
               });
             } else {
               setState({
                 ...state,
                 state: 'error',
-                data: await response.json(),
-                error: `[${response.status}]: ${response.statusText}`,
+                error: {
+                  code: response.status,
+                  message: response.statusText,
+                  ...data,
+                },
```

Comment:
What if we returned the whole Response object?

Reply:
To make the next-runtime less opinionated? Or do you see a use case?

Reply:
It just seemed simpler 🤷 let the user handle the response how they want to, I suppose

Reply:
I'm not a big fan of having `await response.json()` in my projects. I rather have it abstracted away. 

What do you think of:

```js
error: {
  status: response.status,
  statusText: response.statusText,
  headers: response.headers,
  body: await: response.json() || response.text(),
}
```

We could do the same for onSuccess

Reply:
Works for me!